modernizer-annotation-processor/src/main/java/org/gaul/modernizer_annotation_processor/ModernizerAnnotationProcessor.java

```diff
@@ -0,0 +1,341 @@
+        return classHeader;
+    }
+
+    /**
+     * Returns the class name of the element of kind ElementKind.CLASS
+     * passed as a parameter. This includes the names of all the outer classes.
+     * @param classElement The element of kind ElementKind.CLASS whose
+     * class name is to be returned
+     * @return String with $ separated nested class names of the given element
+     * Example:
+     * Output: "ExampleClass$TestClass"
+     */
+    private String getClassName(Element classElement) {
+        List<String> parentClasses = new ArrayList<String>();
+        Element enclosingElement = classElement.getEnclosingElement();
+        while (enclosingElement != null &&
+            enclosingElement.getKind().isClass()) {
+            parentClasses.add(enclosingElement.getSimpleName().toString());
+            enclosingElement = enclosingElement.getEnclosingElement();
+        }
+        StringBuilder className = new StringBuilder();
+        for (int index = parentClasses.size() - 1; index >= 0; index--) {
+            className.append(parentClasses.get(index));
+            className.append("$");
+        }
+        return className.toString() + classElement.getSimpleName();
+    }
+
+    private String getMethodRepresentation(Element methodElement) {
+        ExecutableType method = getExecutableTypeElement(methodElement);
+        return ModernizerAnnotationUtils.getMethodRep(
+        getFullClassName(methodElement.getEnclosingElement()),
+        methodElement.getSimpleName().toString(),
+        getRepresentation(method.getReturnType()),
+        getParamsRepresentation(method.getParameterTypes(), methodElement));
+    }
+
+    private ExecutableType getExecutableTypeElement(Element methodElement) {
+        TypeVisitor<ExecutableType, Void> executableTypeVisitor =
+            new SimpleTypeVisitor6<ExecutableType, Void>() {
+                @Override
+                public ExecutableType visitExecutable(
+                    ExecutableType executableType, Void obj
+                ) {
+                    return executableType;
+                }
+            };
+        return methodElement.asType().accept(executableTypeVisitor, null);
+    }
+
+    /**
+     * Fetches the method parameters in the format the way ASM parses a method
+     * descriptor.
+     * @param methodParams A list of types of formal parameters
+     * of an executable type
+     * @param methodElement The element of executable type
+     * @return A list of all the parameters after formatting
+     * Example:
+     * Input: methodParams {@code {int[][], String, List<Integer>}}
+     * Output: {@code {"int[][]", "String", "java.util.List"}}
+     */
+    private List<String> getParamsRepresentation(
+        List<? extends TypeMirror> methodParams,
+        Element methodElement
+    ) {
+        List<String> methodParamReps = new ArrayList<String>();
+        /*
+         * For a non-static inner class constructor, adding the outer class
```

Comment:
`Add the containing class as a parameter to non-static inner class constructors`